app/services/PsrSubmissionService.scala

```diff
@@ -120,6 +116,31 @@ class PsrSubmissionService @Inject()(
       )
   }
 
+  private def callSubmission(
+    srn: Srn,
+    submissionRequest: PsrSubmission,
+    isSubmitted: Boolean
+  )(implicit hc: HeaderCarrier, ec: ExecutionContext, request: DataRequest[_]): Future[Either[String, Unit]] =
+    if (isPrePopulation(implicitly) && !isSubmitted) {
```

Comment:
we don't need to say (implicitly) here explicitly unless we have another parameter implicitly and provide it explicitly. For example :
```
private def isPrePopulation(implicit request: DataRequest[_], int: explicit): Boolean = ???
isPrePopulation(implicitly, 1)
```

but in our case `if (isPrePopulation && !isSubmitted)`  must be alright.